### DISCUSSION TITLE
Rotate during object drag causes jump

### DISCUSSION THREAD

**Discussion by theredsox** [fabricjs/fabric.js#10014] [General]
[JSFiddle example](https://jsfiddle.net/px74nuwj/3/)

In the JSFiddle example above,
1) left click and hold on the blue rectangle. Optionally drag it around a bit.
2) press 'alt' key to rotate object 90 degrees (while holding left click down)
3) try to continue moving the object around

You'll notice the object doesn't rotate as expected. It jumps around. I think I have narrowed down the issue to the canvas._currentTransform. This gets created when mousedown happens on an object and that transform is used to do the moving (I think). If in the middle of a move, I rotate the object, the _currentTransform isn't being updated to account for that rotation change.

[JSFiddle - workaround](https://jsfiddle.net/xnruftgh/2/)

If you look at the bugFix() function in the workaround JS Fiddle you'll see what I do after the rotation to fix the jumping issue. I imagine there is a better solution that would account for other changes during drag like zoom, skewing...etc. Maybe _currentTransform.original can be compared to _currentTransform.target to determine what has changed and take it into account automatically in places _currentTransform is used. That's beyond my current understanding though.

If anyone has a simpler workaround/solution, I'd be interested to know. Thanks.

**Comment by asturur** [comment]
I would just transition to originX='center' and originY='center' in the project and change the angle instead of calling rotate.

originX and originY are going to be removed anyway and set internally to center.

The use case you demonstrated while is not planned is not completely out of the ordinary, but i don't have an internal smarter solution right now.

Maybe the transform should calculate the offset from the center rather than top/left
